### ISSUE TITLE
Add documentation about known issue regarding xunit tests when targeting .NET Full Framework

### ISSUE THREAD

**Issue by AdmiringWorm** [cake-contrib/Cake.Recipe#371]
When trying to run unit test on Linux, and you have either is using multiple target frameworks, or is just targeting a Full .NET Framework, then the .NET Core test utility is unable to actually find/run the unit tests (It will still run unit tests for .NET Core and .NET Standard).

I know this is a known problem in the .NET Core repository, and the reason why this happen can be different between different linux distros (in some it may actually succeed as well). In my case, it is because the testhost adapter that .NET Core tries to run, don't exist anywhere (and no way of getting a hold of it).

As such, I suggest that when running on a Unix platform, run only .NET Core targeted frameworks using `dotnet test`. But run the correct unit test executable for frameworks targeting the full .NET Framework (like what is done when running a normal MSBuild).

**Comment by gep13** [comment]
@AdmiringWorm is this still an issue?

**Comment by AdmiringWorm** [comment]
I think it is, but haven't really tested it since I created the issue.
In my projects I am disabling full framework tests on Unix platforms.

**Comment by AdmiringWorm** [comment]
Just added .NET Framework to the .NET Core unit tests, and it seems to be still an issue.

At the very least xUnit fails for .NET Framework, and it looks like there is something wrong for NUnit tests as well (although no failure here).
https://travis-ci.org/github/AdmiringWorm/Cake.Recipe.Tests/jobs/707742236#L3089

**Comment by gep13** [comment]
Doh! 😢

**Comment by AdmiringWorm** [comment]
On a positive note though, MSBuild unit tests seems to work out-of-the-box 😄

**Comment by gep13** [comment]
Woot! 🎉

**Comment by AdmiringWorm** [comment]
By doing some different testing regarding this issue, I found that there isn't much we can/should do in cake.recipe itself. But I believe this is an issue we would need/want to document (maybe in a known issues section?).

Now onto what the issue actually is:

The issue seems to only be related to the xunit testing framework when targeting a .NET Full Framework version and running on Unix (not confirmed if it is just Linux, or if macosx is also affected, but I believe so).
For the package, there is an implicit dependency on [`Microsoft.TestPlatform.ObjectModel`](https://nuget.org/packages/Microsoft.TestPlatform.ObjectModel) which gets pulled in automatically when running on windows, but unfortunately do not on Unix.
Because of this, this dependency is needed to be explicitly referenced in the unit testing project.
The best way I found to do this, was to only restore the dependency when running .NET Full Framework and on Unix with the following csproj addition.
```xml
<PackageReference Include="Microsoft.TestPlatform.ObjectModel" Version="16.6.1" Condition="$(TargetFramework.StartsWith('net4')) AND '$(OS)' == 'Unix'" />
```

The above package reference version should also match the package reference version of [`Microsoft.NET.Test.Sdk`](https://nuget.org/packages/Microsoft.NET.Test.Sdk)

The error text that a user gets without this package reference may look similar to:
```
[xUnit.net 00:00:00.76] XUnitTests: Catastrophic failure: System.TypeLoadException: Could not load type of field 'Xunit.Runner.VisualStudio.VsExecutionSink:recorder' (4) due to: Could not load file or assembly 'Microsoft.VisualStudio.TestPlatform.ObjectModel, Version=11.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies.
```
